src/components/Table/Table.js

```diff
@@ -50,49 +64,38 @@ class DataTable extends React.Component {
         return fetch(`https://cors-anywhere.herokuapp.com/http://18.224.29.151:5000/get-stops?stop_code=${R.join(",", stops)}`)
             .then(res => res.json())
             .then(res => R.map(stop => R.includes({ stop_code: stop }) ? res[R.findIndex(R.propEq('stop_code', stop))(res)].stop_name : "Geen haltenaam beschikbaar", stops))
+
     }
 
     formatTime(item) {
-        let minutes = Math.floor(item / 60).toString()
-        let seconds = (item % 60).toString()
+        const minutes = Math.floor(item / 60).toString()
+        const seconds = Math.round(item % 60).toString()
 
-        return (minutes >= 1 ? minutes + " minuten en " : "") + seconds +  " seconden"
+        return (minutes >= 1 ? minutes + " minuten en " : "") + seconds + " seconden"
       }
 
     render() {
         if (this.state.values == null) {
             return <Missing/>
         }
 
-        return (
-            <table className="striped">
-                <tbody>
-                    <tr className="table-header-row">
-                        {this.state.headers.map(col => <th key={col} className="table-header-row-value">{col}</th>)}
-                    </tr>
-
-                    {
-                        this.state.values.map((row, i) => {
-                            var metric = row.metric
-                            var values = row.value
-
-                            return <tr className="table-row" key={i}>
-                                <td className="table-row-value" key={i}>{i + 1}</td>
-                                {
-
-                                    this.props.order.map(val => {
-                                        return <td className="table-row-value" key={metric[val]}>{metric[val]}</td>
-                                    })
-                                }
-
-                                <td className="table-row-value" key={values[1]}>{this.formatTime(parseInt(values[1]))}</td>
-                            </tr>
-                        })
-                    }
-
-                </tbody>
-            </table>
-      )
+        /* Creates the table header and binds the rows to the data in the columns
+         * using the accessor. Also manually sets width for first row. */
+        const columns = this.state.headers.map(h => ({Header: h, accessor: h}))
+        columns[0] = {maxWidth: 100, ...columns[0]}
```

Comment:
I dont like this at all, but I couldn't find a more convenient way that still uses map.

Reply:
why only the first element?

Reply:
First element is always the "Nr" column, but i've changed it now to a function that chooses width based on the length.